src/main/java/com/cleanroommc/groovyscript/compat/mods/astralsorcery/Constellation.java

```diff
@@ -0,0 +1,248 @@
+        if (ConstellationRegistryAccessor.getConstellationList() == null) return;
+        if (doBackup) this.addBackup(constellation);
+        this.removeConstellationMapEffect(constellation, doBackup);
+        ConstellationRegistryAccessor.getConstellationList().removeIf(registeredConstellation -> constellation.getSimpleName().equals(registeredConstellation.getSimpleName()));
+        if (constellation instanceof IMajorConstellation && ConstellationRegistryAccessor.getMajorConstellations() != null)
+            ConstellationRegistryAccessor.getMajorConstellations().removeIf(registeredConstellation -> constellation.getSimpleName().equals(registeredConstellation.getSimpleName()));
+        if (constellation instanceof IMinorConstellation && ConstellationRegistryAccessor.getMinorConstellations() != null)
+            ConstellationRegistryAccessor.getMinorConstellations().removeIf(registeredConstellation -> constellation.getSimpleName().equals(registeredConstellation.getSimpleName()));
+        if (constellation instanceof IWeakConstellation && ConstellationRegistryAccessor.getWeakConstellations() != null)
+            ConstellationRegistryAccessor.getWeakConstellations().removeIf(registeredConstellation -> constellation.getSimpleName().equals(registeredConstellation.getSimpleName()));
+    }
+
+    private void addConstellationMapEffect(IConstellation constellation, List<ConstellationMapEffectRegistry.EnchantmentMapEffect> enchantmentEffectList, List<ConstellationMapEffectRegistry.PotionMapEffect> potionEffectList) {
+        this.constellationMapEffectsAdded.put(constellation, ConstellationMapEffectRegistry.registerMapEffect(constellation, enchantmentEffectList, potionEffectList));
+    }
+
+    public void removeConstellationMapEffect(IConstellation constellation) {
+        this.removeConstellationMapEffect(constellation, true);
+    }
+
+    private void removeConstellationMapEffect(IConstellation constellation, boolean doBackup) {
+        if (ConstellationMapEffectRegistryAccessor.getEffectRegistry() == null) return;
+        if (ConstellationMapEffectRegistryAccessor.getEffectRegistry().containsKey(constellation)) {
+            if (doBackup) this.constellationMapEffectsRemoved.put(constellation, ConstellationMapEffectRegistryAccessor.getEffectRegistry().get(constellation));
+            ConstellationMapEffectRegistryAccessor.getEffectRegistry().remove(constellation);
+        }
+    }
+
+    public ConstellationBuilder constellationBuilder() {
+        return new ConstellationBuilder();
+    }
+
+    public ConstellationMapEffectBuilder constellationMapEffectBuilder() {
+        return new ConstellationMapEffectBuilder();
+    }
+
+    public static class ConstellationBuilder {
+
+        private String name;
+        private Color color = null;
+        private ConstellationBuilder.Type type;
+        private final ArrayList<Tuple<Point,Point>> connections = new ArrayList<>();
```

Comment:
Either use `org.apache.commons.lang3.tuple.Pair` instead of `Tuple` or make new class with 4 ints.